### DISCUSSION TITLE
High Availability

### DISCUSSION THREAD

**Discussion by seang96** [ciur/papermerge#616] [Q&A] [answered]
This appears to work well with high availability with reading the docs, design, but nothing explicit including the kurbenetes setup not ironed out. Is this possible currently? It looks like all the backend services can be clustered and the worker according to the docs, but everything says web frontend to have only 1 instance. Is there something extra the front end service does besides being a user endpoint? Below is the tech stack I am planning on an attempt for.

1. Host multiple of the web frontend load balanced by proxy server
2. Host multiple workers
3. Postgres - cloudnative postgres operator
4. Solr - Solr operator
5. Redis - dragonfly operator

Thanks!

**Comment by ciur** [comment, marked as answer]
This is what I am currently working on. The 3.3 release is about changes to accomodate k8s deployments. 

The helm charts are in  https://github.com/papermerge/helm-chart

repo.

As I mentioned, it is work in progress - both helm chart and the documentation.
The k8s deployment will work only with 3.3 onwards.

**Reply by seang96** [reply]
I managed to get it running with the setup outlined. Ironically the biggest problem was integration with keycloak auth.

- refresh tokens aren't used and my token auth expires quickly. I upped my expiration time for the client.
- ingress-nginx proxy needed its request buffer increased to support logging in with the "groups" scope added.

**Reply by ciur** [reply]
yes, you are right, the oidc support in clunky. I will be fixed in future.
The 3.3 will come with "cloud" version i.e. there will be 3.3-cloud docker tag  (besides usual 3.3) which will have much better integration with keycloak. But as I mentioned it is work in progress.

**Comment by ciur** [comment]
btw, thanks for dragonfly tip. I didn't know about it; I googled it... looks very promising! And there is an k8s operator as well !I will give it a try.

**Reply by seang96** [reply]
No problem! It looks like all the operators are working well in my stack but I unfortunately lack a scanner so I couldn't test this much.
